### ISSUE TITLE
JEI Bookmarks Losing Pages

### ISSUE THREAD

**Issue by no-name-provided** [FTBTeam/FTB-Modpack-Issues#6520]
### Modpack

FTB Evolution

### Modpack version

1.6.0

### Launcher

FTB App

### Has the pack been modified

Yes

### Describe the bug

Under some circumstances, JEI forgets about inactive pages of bookmarks. It eventually remembers them, but you need to escape out of JEI recipe lookup to make this happen.

### Steps to reproduce

Create enough bookmarks to overflow one page.
Open a GUI (eg, crystal assembler)
Navigate to second page and left click first recipe

### Expected behaviour

Favorites to remain accessible

### Screenshots

![image](https://github.com/user-attachments/assets/98f37854-7dfa-4ddf-b720-86e233c957b1)

![image](https://github.com/user-attachments/assets/60359d9d-9475-437a-a828-200cf109cf6c)

### Additional information

Again, using alltheleaks.

**Comment by Gaz492** [comment]
If you remove alltheleaks, does the issue continue to occur?
From a quick look at the alltheleaks mod it looks like it touches the bookmark code in jei so it may be the cause of the issues here.

**Comment by no-name-provided** [comment]
Yes, it does. Actually, the problem gets worse. Opening a GUI, then switching to the second page, then closing the GUI now "deletes" the first page of bookmarks until I open another GUI.

Are you having trouble replicating this?

**Comment by Gaz492** [comment]
Unfortunately, I haven't been able to reproduce this, I'd possibly suggest opening an issue on the JEI issue tracker and see what they have to say, it might be the case where the issue is only reproducible in a very specific scenario that I am unable to meet.
https://github.com/mezz/JustEnoughItems/issues

**Comment by no-name-provided** [comment]
Did you follow the steps for reproduction exactly? Set GUI scale to 3, have enough bookmarks to have two pages while looking at crystal assembler GUI, but only one page in regular inventory, open large GUI (eg, crystal assembler) and switch to BM page two, then close GUI and open inventory? Click arrow buttons on top of bookmarks area and observe no change, while 1/1 is displayed as the page count?

This isn't an intermittent issue.

**Comment by Griese009** [comment]
I had the issue too once, a possible fix would be to delete each item on the current page, the other pages should reappear then.

Yes i know that this is more likely a temporary fix.

**Comment by Gaz492** [comment]
> Did you follow the steps for reproduction exactly? Set GUI scale to 3, have enough bookmarks to have two pages while looking at crystal assembler GUI, but only one page in regular inventory, open large GUI (eg, crystal assembler) and switch to BM page two, then close GUI and open inventory? Click arrow buttons on top of bookmarks area and observe no change, while 1/1 is displayed as the page count?
> 
> This isn't an intermittent issue.

So I managed to reproduce this once while playing around with the window size as using the settings you mentioned didn't allow me to reproduce this, however I'm not always able to reproduce this as I'm having issues with making sure that the bookmarked items are on one page while in the regular inventory and with one item on a second page while in the crystal assembler.

So I do believe this is a JEI issue and as I am unable to consistently and easily reproduce this issue. I believe it's best that you report the issue to the JEI developer https://github.com/mezz/JustEnoughItems/issues as if they require any further debugging steps, it seems your PC setup allows you to replicate this consistently.

**Comment by no-name-provided** [comment]
Reminder: the reply the bot's awaiting is yours, Gaz492.
